client/post-editor/editor-copy-post/index.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * Internal dependencies
+ */
+import actions from 'lib/posts/actions';
+import { editPost } from 'state/posts/actions';
+import { setEditorPostId } from 'state/ui/editor/actions';
+import wpcom from 'lib/wp';
+
+function setPostCopy( post ) {
+	const postCopy = {
+		options: {
+			type: 'post',
+			content: post.content,
+		},
+		attributes: {
+			categories: Object.keys( post.categories ).map( key => post.categories[ key ].ID ),
+			excerpt: post.excerpt,
+			tags: Object.keys( post.tags ).map( key => post.tags[ key ].name ),
+			title: post.title,
+		},
+	};
+
+	if ( post.canonical_image ) {
```

Comment:
Maybe Lodash `_.pick` here? Would omit the non-existent attributes:

```
postCopy.attributes = pick( post, 'canonical_image', 'featured_image', 'format', 'metadata', 'post_thumbnail' );
```

https://lodash.com/docs#pick